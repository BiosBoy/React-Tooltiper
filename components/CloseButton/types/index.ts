import { ICloseButton } from '../../../types'

export interface IProps extends ICloseButton {
  clickAction?: () => void
}
