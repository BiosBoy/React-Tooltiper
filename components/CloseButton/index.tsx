import React, { PureComponent } from 'react'

import { IProps } from './types'

class CloseButton extends PureComponent<IProps> {
  _handleClick = e => {
    const { clickAction } = this.props

    if (!clickAction) return

    clickAction()

    // shold not affect events during bubbling phase
    e.preventDefault()
    e.stopPropagation()
  }

  render() {
    const { active, buttonChildren, buttonStyles } = this.props

    if (!active) {
      return null
    }

    return (
      <button type='button' className={buttonStyles} onClick={this._handleClick}>
        {buttonChildren || 'X'}
      </button>
    )
  }
}

export default CloseButton
