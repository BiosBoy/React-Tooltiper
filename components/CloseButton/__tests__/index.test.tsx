import React from 'react'
import { mount } from 'enzyme'

import initialState, { disabled } from './mocks'
import CloseButton from '..'

describe('<CloseButton Arrow test />', () => {
  it('should render basic CloseButton', () => {
    const Component = mount(<CloseButton {...initialState} />)

    expect(Component.find('CloseButton').length).toBe(1)
    expect(Component.find('.closeButton').length).toBe(1)

    Component.find('.closeButton').simulate('click')

    expect(Component).toMatchSnapshot()
  })
  it('should not render CloseButton in case of active flag is not provided ', () => {
    const Component = mount(<CloseButton {...disabled} />)

    expect(Component.find('CloseButton').length).toBe(1)
    expect(Component.find('.closeButton').length).toBe(0)

    expect(Component).toMatchSnapshot()
  })
})
