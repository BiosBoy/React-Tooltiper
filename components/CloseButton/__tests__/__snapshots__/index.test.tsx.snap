// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton Arrow test /> should not render CloseButton in case of active flag is not provided  1`] = `
<CloseButton
  active={false}
  buttonStyles="closeButton"
  clickAction={[Function]}
/>
`;

exports[`<CloseButton Arrow test /> should render basic CloseButton 1`] = `
<CloseButton
  active={true}
  buttonStyles="closeButton"
  clickAction={[Function]}
>
  <button
    className="closeButton"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
</CloseButton>
`;
